### PR TITLE
Add interactive overwrite when duplicates are present

### PR DESCRIPTION
**Pull request by dylnclrk**
We want to be able to overwrite files when .example files change, but
we don't want to always overwrite them. Make Exampler interactive to remedy.

### CODE REVIEW

File: exampler [hunk 1]
```diff
@@ -16,6 +17,33 @@ module Exampler
         process_file!(file)
       end
     end
+    def process_file!(file_path)
```
Comment (davidjairala) [inline exampler]: nitpick: linebreak between the methods?

File: exampler [hunk 2]
```diff
@@ -27,22 +55,22 @@ module Exampler
...
-        return
-      end
+    def copy(from, to)
+      %x[cp #{from} #{to}]
+      puts "** copied #{from} -- to -- #{to}"
+    end
 
-      %x[cp #{file_path} #{copy_path}]
-      puts "** copied #{file_path} -- to -- #{copy_path}"
+    def prompt_yes_no(prompt)
```
Comment (davidjairala) [inline exampler]: nit: since bool return method should end with `?`

File: exampler [hunk 3]
```diff
@@ -16,6 +17,33 @@ module Exampler
...
+          return
+        else
+          puts "** file already exists…"
+          puts "** here's the diff:"
+
+          system "git diff --no-index #{copy_path} #{file_path}"
+        end
+
+        if prompt_yes_no("[?] overwrite [#{copy_path}] with new [#{file_path}]?")
+          "** OK! overwriting [#{copy_path}]"
```
Comment (davidjairala) [inline exampler]: missing a `puts`?

**Review by davidjairala — COMMENTED** [review]
Thank you for the PR.  Looks solid.  Some comments.  Lets also maybe add a note to the `README` that we now require `git` to be installed?  I realize that's probably kind of a given, but you know.

File: exampler (hunk 3 above)
Comment (dylnclrk) [reply exampler]: haha 🤦‍♂️

**Comment by dylnclrk** [comment]
Hey David, thanks for the feedback! Apologies for the sloppiness… I quickly threw this feature in this morning. This should be better now. Let me know if I missed anything.

PS I amended my original commit, I hope that's ok.

**Comment by davidjairala** [comment]
Looks great. Thanks.

**Pull request merged**